functions/tokens/[[index]].ts

```diff
@@ -2,17 +2,28 @@
 import { MetaTagInjector } from '../components/metaTagInjector'
 import getToken from '../utils/getToken'
 
-const convertTokenAddress = (tokenAddress: string) => {
-  return tokenAddress && tokenAddress === 'NATIVE' ? '0x0000000000000000000000000000000000000000' : tokenAddress
+const convertTokenAddress = (tokenAddress: string, networkName: string) => {
+  if (tokenAddress === 'NATIVE') {
+    switch (networkName) {
+      case 'CELO':
+        return '0x471EcE3750Da237f93B8E339c536989b8978a438'
+      case 'POLYGON':
+        return '0x0000000000000000000000000000000000001010'
```

Comment:
Can you check with the backend team that this is comprehensive? What about their testnets?

Reply:
We also have a map of wrapped native currencies `WRAPPED_NATIVE_CURRENCY` in `src/constants/tokens.ts` we could leverage

Reply:
@cbachmeier yeah i took a look at that, however, we don't necessarily want the `WRAPPED_NATIVE_CURRENCY`. For `Celo` its fine since they're the same, however, `Polygon` has a wrapped native currency `wMatic` and the native currency `Matic` and `WRAPPED_NATIVE_CURRENCY` would return the address for `wMatic` instead of `Matic`. I don't know if this is an edge case or not though.

Reply:
I was looking at `getNativeTokenDBAddress` in `src/constants/nativeTokens.ts` but saw that only `Celo` and `Polygon` use a different native token which is why I just opted to hard code those 2 chains